### PR TITLE
Customer get by password and email token

### LINKED ISSUE #1448: Add support for get customer by email token
Our backend has a new endpoint to retrieve the customer by email token `/{projectKey}/customers/email-token={token}`. 

Documentation can be found in our documentation repository.

### LINKED ISSUE #1449: Adapt get customer by password token
Together with #1448 our backend changed the endpoint for retrieving the customer by email token (`/{projectKey}/customers/password-token={token}`). We should adapt our code to use this new endpoint.

More information can be found in our documentation github PR.

### PR DESCRIPTION
**Pull request by katmatt**
This closes #1448, closes #1449.

### CODE REVIEW

File: sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            result.add(matcher.group(1));
+        }
+        return result;
+    }
+
+    /**
+     * Returns true if this uri template has the given parameter.
+     *
+     * @param name the parameter name
+     * @return true iff. this uri template has the given parameter
```
Comment (acbeni) [inline sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java]: Just a little mistake 
`* @return true *iff.* this uri template has the given parameter`

File: sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
```
Comment (acbeni) [inline sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java]: Just a little typo :simple_smile: `true *iff.* this`

Comment (katmatt) [inline sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java]: It's not a typo, it's an acronym for ([if and only if](https://en.wikipedia.org/wiki/If_and_only_if)) 😄

**Comment by katmatt** [comment]
And thanks and yes, we should use the new uri helpers in more places. And improve them further 😉

File: sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java (hunk 2 above)
Comment (acbeni) [inline sdk-http/src/main/java/io/sphere/sdk/http/UriTemplate.java]: Oh my :sweat_smile: than its good i guess :open_mouth:

**Pull request merged**